phpBB/phpbb/cron/task/wrapper.php

```diff
@@ -105,6 +105,27 @@ public function get_url()
 		return $this->routing_helper->route('phpbb_cron_run', $params);
 	}
 
+	/**
+	 * Returns HTML for an invisible `img` tag that can be displayed on page
+	 * load to trigger a request to the relevant cron task endpoint.
+	 *
+	 * @return string       HTML to render to trigger cron task
+	 */
+	public function get_html_tag()
+	{
+		global $template;
```

Comment:
This should be removed. No globals.

Reply:
> This should be removed. No globals.

What's the correct way of accessing `$template` here then? I noticed a common pattern is this:

```php
global $phpbb_container;

/** @var \phpbb\template\template $template Template object */
$template = $phpbb_container->get('template');
```

However, that simply pushes the problem one step down the line, as you still need the `global $phpbb_container;`. Or is `$phpbb_container` allowed as a global?

Reply:
`use phpbb\template\template;`

_then_

`protected $template`

_then in the constructor_

`public function __construct(task $task, helper $routing_helper, $phpbb_root_path, $php_ext, template $template)`

_then_

`$this->template	= $template;`

**NOT TESTED.**

Reply:
Fixed with dependency injection (doesn't need `use phpbb\template\template;`).

Reply:
> Fixed with dependency injection (doesn't need `use phpbb\template\template;`).

I wrote on the fly just to give you a starter, well done.